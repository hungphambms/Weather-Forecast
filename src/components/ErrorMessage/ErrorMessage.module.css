.message {
    font-size: 16px;
    line-height: var(--line-height);

    color: var(--color-medium);
    margin: 16px 0;
    padding: 0;
}