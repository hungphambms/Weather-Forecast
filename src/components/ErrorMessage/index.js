import PropTypes from "prop-types";

import styles from "./ErrorMessage.module.css";

function ErrorMessage({ message }) {
  return (
    <div aria-label="error-message" className={styles.message}>
      {message}
    </div>
  );
}

ErrorMessage.propTypes = {
  message: PropTypes.string.isRequired,
};

export default ErrorMessage;
