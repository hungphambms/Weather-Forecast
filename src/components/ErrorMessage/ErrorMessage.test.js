import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

import ErrorMessage from "./index";

const MESSAGE_PROPS = { message: "Have error" };

describe("ErrorMessage rendering", () => {
  it("renders ErrorMessage component without crashing", () => {
    const component = renderer.create(<ErrorMessage {...MESSAGE_PROPS} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("show correct error message", () => {
    const { getByText, unmount } = render(<ErrorMessage {...MESSAGE_PROPS} />);

    const headingContainer = getByText(MESSAGE_PROPS.message);
    expect(headingContainer.textContent).toBe(MESSAGE_PROPS.message);

    unmount();
  });
});
