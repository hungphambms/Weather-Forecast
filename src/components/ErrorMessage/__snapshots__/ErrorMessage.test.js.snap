// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage rendering renders ErrorMessage component without crashing 1`] = `
<div
  aria-label="error-message"
  className="message"
>
  Have error
</div>
`;
